### ISSUE TITLE
`quarkus.otel.traces.exporter` and `quarkus.otel.traces.sampler` should be runtime config

### ISSUE THREAD

**Issue by jbonofre** [quarkusio/quarkus#36471]
### Describe the bug

Currently, `quarkus.otel.traces.exporter` and `quarkus.otel.traces.sampler` configurations are build time.

They should be runtime in order to give a chance to SRE team to customize/update the otel exporter/sampler configuration.

### Expected behavior

`quarkus.otel.traces.exporter` and `quarkus.otel.traces.sampler` should be runtime config.

### Actual behavior

`quarkus.otel.traces.exporter` and `quarkus.otel.traces.sampler` are buildtime config.

**Comment by jbonofre** [comment]
@brunobat thoughts ?

**Comment by jbonofre** [comment]
@brunobat I will work on a PR if it works for you.

**Comment by brunobat** [comment]
I would like to have the bean resolution for these OTel SPIs to remain at build time. We already moved a lot of bootstrap logic for runtime. All this has start time impact and we decided that doing the cut on the SPIs would be the best compromise.
@jbonofre feel free to work on a proposal... Nothing better than test what will happen.

**Comment by jbonofre** [comment]
The reason is to allow my SREs to change the otel exporter without the need of rebuilding the app (just restarting the pod). The SREs "control" the exporter at runtime, the dev team doesn't know the exporter at build time.

**Comment by brunobat** [comment]
Yes, that is the usual reason, but is there a specific request from someone for this use case?
